tree/ntuple/v7/src/RField.cxx

```diff
@@ -2498,6 +2498,17 @@ void ROOT::Experimental::RProxiedCollectionField::AcceptVisitor(Detail::RFieldVi
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RRecordField::RRecordField(std::string_view name, const RRecordField &source)
+   : ROOT::Experimental::RFieldBase(name, source.GetTypeName(), ENTupleStructure::kRecord, false /* isSimple */),
+     fMaxAlignment(source.fMaxAlignment),
+     fSize(source.fSize),
+     fOffsets(source.fOffsets)
+{
+   for (const auto &f : source.GetSubFields())
+      Attach(f->Clone(f->GetFieldName()));
+   fTraits = source.fTraits;
```

Comment:
We might want to move this to the generic `RFieldBase::Clone` because it's a base class member

Reply:
Indeed. I was under the impression that traits could change when a field gets connected (e.g., in the presence of read rules). Clone resets the connection status. However, traits are not changed with the connection status.